src/main/include/log4cxx/spi/location/locationinfo.h

```diff
@@ -45,21 +58,28 @@ class LOG4CXX_EXPORT LocationInfo
 
 		static const LocationInfo& getLocationUnavailable();
 
-
+#ifdef LOG4CXX_HAS_STRING_VIEW
+		static constexpr const char* calcShortFileName(const char* fileName){
+			std::string_view view(fileName);
+			// If the separator is not found, rfind will return -1.  Adding 1 to
+			// that will have it pointing at fileName, which is a good fallback.
+			return fileName + view.rfind(LOG4CXX_SHORT_FILENAME_SPLIT_CHAR) + 1;
+		}
+#else
+		static const char* calcShortFileName(const char* fileName){
```

Comment:
Some additional docs of why `constexpr` is not possible with `strrchr`:

https://stackoverflow.com/questions/42352114/why-all-function-in-cstring-must-not-have-constexpr